src/azure-cli/azure/cli/command_modules/appconfig/_validators.py

```diff
@@ -159,3 +162,32 @@ def validate_secret_identifier(namespace):
         KeyVaultIdentifier(uri=identifier)
     except Exception as e:
         raise CLIError("Received an exception while validating the format of secret identifier.\n{0}".format(str(e)))
+
+
+def validate_key(namespace):
+    if namespace.key:
+        input_key = str(namespace.key)
+        if input_key == '.' or input_key == '..' or '%' in input_key:
+            raise CLIError("Key is invalid. Key cannot be a '.' or '..', or contain the '%' character.")
```

Comment:
why we need to check for these three cases? It makes sense to make sure key shouldn't start with feature flag prefix. But I am not sure why we have a shallow check for some characters here.

Reply:
To be consistent with all the checks in portal.

---
In reply to: [362633661](https://github.com/Azure/azure-cli/pull/11753#discussion_r362633661) [](ancestors = 362633661)